readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -1476,6 +1476,9 @@ def test_save_environment_json(self, load_config):
             python_env.save_environment_json()
             json_data = json.load(open(python_env.environment_json_path()))
 
+        envvars = self.version.project.environmentvariable_set.values_list('name', 'value')
+        envvars_hash = hash(tuple(envvars))
```

Comment:
I don't think this is the `hash` function we need here.

https://docs.python.org/3/library/functions.html#hash

It seems that it get the hash of the object, but I'm not sure if it's based on the `name` and `value` values. I suppose that we need to use another function for this.

Reply:
I didn't get what you are trying to say.
I have pushed some new changes.

Reply:
Confirmed. This is not what we want here since it will change from one run to another even having the same values.

```
In [10]: hash(tuple(EnvironmentVariable.objects.all().values_list('name', 'value')))                                                                                                          
Out[10]: 1936855537803375997
```

Close the interpreter and open it again,

```
In [1]: hash(tuple(EnvironmentVariable.objects.all().values_list('name', 'value')))                                                                                                           
Out[1]: -3229000353354874828
```

We need to use "real" hash function that consider the `name` and `value` to calculate it (the contents) instead of the internal Python representation of the object.

Take a look at `hashlib` module for example, and sha256: https://docs.python.org/3/library/hashlib.html

Reply:
Thanks @humitos for this.
I have updated the PR.